projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js

```diff
@@ -18,7 +21,23 @@ export default function SubscribePanels() {
 		getSubscriberCount( count => setSubscriberCount( count ) );
 	}, [] );
 
-	return ! Number.isFinite( subscriberCount ) || subscriberCount <= 0 ? null : (
+	// Only show this for posts for now (subscriptions are only available on posts).
+	const postType = useSelect( select => select( editorStore ).getCurrentPostType(), [] );
+	if ( 'post' !== postType ) {
+		return null;
+	}
+
+	// Subscriptions will not be triggered on WoA's private sites.
+	if ( isAtomicSite() && isPrivateSite() ) {
```

Comment:
Turns out that `isPrivateSite()` returns `false` when the site hasn't been launched. 🤔 

As a result, the text still displays in the sidebar.

Reply:
On the PHP side we have `is_coming_soon()` and `get_launch_status()` available, but I haven't found an equivalent on the JS/React side yet.

Reply:
Ah, I guess I'm not testing "not launched" properly then; I was just making my site private under Settings > General, but that's different from "not launched" I see. I'll come up with something to fix not launched sites as well.

Reply:
Yeah I had tested it by setting my site to private. Strange that I'd expect isPrivateSite() to return true. I see we have an endpoint for the launch status - maybe helpful?  https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/sal/class.json-api-site-jetpack.php#L185

Reply:
Just pushed f502a5423a9a2c1548e9a1b6b9d77dce4c7d5c81 to try to address this situation.

Reply:
I just spent some time testing this now using a non-launched WoA site, using this feature branch with the beta plugin. The subscription panel does show for posts only, in the pre-publish panel.
<img width="264" alt="Screenshot 2022-03-31 at 11 11 20" src="https://user-images.githubusercontent.com/16754605/161034693-0a35db1d-d8fa-46bc-a180-216ca17ea8ec.png">

That was my first time setting up a WoA dev blog so it's possible I've missed something, but everything seems to be in place from what I can see.

Reply:
I've just pushed 6e1fcf5a2c1c6cb49f2d314973767c6ab70d42cd to try a fix, I had missed that export so my last changes were not building properly, sorry!

Reply:
Works as expected now. 😊👍🏼 

When the site hasn't been launched, the "Subscribe" panel doesn't load (it doesn't load when publishing a _page_ either):
![Markup on 2022-03-31 at 12:54:13](https://user-images.githubusercontent.com/25105483/161052222-73f387c8-1ac1-4461-a994-50aaaa294928.png)

As soon as the site is launched and _public_, the "Subscribe" panel loads when publishing a _post_:
![Markup on 2022-03-31 at 12:56:16](https://user-images.githubusercontent.com/25105483/161052304-732318d8-a7b2-4ab1-b3fb-bfd8e79a2ce9.png)

Tested the changes when the site was in the _coming soon_ and _private_ mode as well. The "Subscribe" panel doesn't load - as expected.

Reply:
That works for me now too. However after launching the site, with the PR being tested there is no subscriber panel pre-publish with posts, and when I remove the beta plugin there is a subscriber panel showing.
It may just be this WoA site though - despite being public now every WordPress.com dashboard load shows it momentarily as being in 'Coming Soon' mode, then that disappears.

Reply:
> [...] despite being public now every WordPress.com dashboard load shows it momentarily as being in 'Coming Soon' mode, then that disappears.

Interesting. Does it appear and disappear for you in the "pre-publish" panel? 🤔

Reply:
Ah no, sorry. I meant the 'coming soon' banner showing under the site URL (under 'My Sites') in the WordPress.com dashboard appears and then disappears. However I just changed it back to private then back to public again, and now everything works perfectly 🤷